src/framework/app-base.js

```diff
@@ -522,7 +524,7 @@ class AppBase extends EventHandler {
          *     // VR is available
          * }
          */
-        this.xr = new XrManager(this);
+        this.xr = new createOptions.xr(this);
```

Comment:
This feels weird. Why not just pass an `XrManager` instance to the app?

Reply:
most of the others are done the same way - we just assign the type, but not constructed object.

```
        createOptions.lightmapper = Lightmapper;
        createOptions.batchManager = BatchManager;

        createOptions.componentSystems = [
            RigidBodyComponentSystem,
            CollisionComponentSystem,
            JointComponentSystem,
           ...
```

this allows us to hide constructor / modify constructor parameters in the future without affecting user's code. And allows us to instantiate those in some fixed order we need.

Reply:
I think it works nicely for component systems and resource handles. But yeah, feels weird for the `XrManager` and `Lightmapper`.

Reply:
makes sense .. I felt strange about it .. especially since GraphicsDevice and SoundManager are created using new, and others not. I'll unify that soon, and use `new` on all those.